packages/sveltekit/src/cookie-storage.test.ts

```diff
@@ -0,0 +1,175 @@
+      httpOnly: true,
+      path: '/',
+      sameSite: 'lax',
+      secure: false,
+      maxAge: 14 * 24 * 3600,
+    });
+
+    expect(
+      new TestCookieStorage({
+        requestEvent: createMockEvent({}, 'https://localhost/'),
+        encryptionKey,
+      }).getCookieOptions()
+    ).toEqual({
+      httpOnly: true,
+      path: '/',
+      sameSite: 'lax',
+      secure: true,
+      maxAge: 14 * 24 * 3600,
+    });
+
+    expect(
+      new TestCookieStorage({
+        requestEvent: createMockEvent({}, undefined, new Map([['x-forwarded-proto', 'https']])),
+        encryptionKey,
+      }).getCookieOptions()
+    ).toEqual({
+      httpOnly: true,
+      path: '/',
+      sameSite: 'lax',
+      secure: true,
+      maxAge: 14 * 24 * 3600,
+    });
+  });
+
+  it('should be able to load and save data', async () => {
+    const requestEvent = createMockEvent({
+      logtoCookies: await wrapSession({ [PersistKey.AccessToken]: 'bar' }, encryptionKey),
+    });
+    const storage = new TestCookieStorage({ requestEvent, encryptionKey });
+    await storage.init();
+    expect(storage.data).toEqual({ [PersistKey.AccessToken]: 'bar' });
+    expect(await storage.getItem(PersistKey.AccessToken)).toEqual('bar');
+    expect(await storage.getItem(PersistKey.IdToken)).toBeNull();
+
+    await storage.setItem(PersistKey.AccessToken, 'baz');
+    expect(storage.data).toEqual({ [PersistKey.AccessToken]: 'baz' });
+    expect(await unwrapSession(requestEvent.cookies.get('logtoCookies')!, encryptionKey)).toEqual({
+      [PersistKey.AccessToken]: 'baz',
+    });
+  });
+
+  it('should be able to remove data', async () => {
+    const requestEvent = createMockEvent({
+      foo: await wrapSession({ [PersistKey.AccessToken]: 'bar' }, encryptionKey),
+    });
+    const storage = new TestCookieStorage({ requestEvent, encryptionKey, cookieKey: 'foo' });
+    await storage.init();
+    expect(storage.data).toEqual({ [PersistKey.AccessToken]: 'bar' });
+
+    await storage.removeItem(PersistKey.AccessToken);
+    expect(storage.data).toEqual({});
+    expect(await unwrapSession(requestEvent.cookies.get('foo')!, encryptionKey)).toEqual({});
+  });
+});
+
+describe('CookieStorage concurrency', () => {
+  class ConcurrentTestCookieStorage extends TestCookieStorage {
+    // Simulate encryption delay for testing concurrent updates
+    protected override async write(): Promise<void> {
+      const staleSessionData = { ...this.sessionData };
+
+      return delay(
+        async () => {
+          await super.write(staleSessionData);
+        },
+        this.sessionData.idToken ? 0 : 5 // Ensure other updates are slower than idToken
+      );
+    }
+  }
+
+  class NoQueueTestCookieStorage extends ConcurrentTestCookieStorage {
+    protected override async save(): Promise<void> {
+      return this.write();
+    }
+  }
+
+  // For each storage class, test three times to ensure the result is consistent with fixed
+  // conditions and random delays.
```

Comment:
@wangsijie this should happen in next sdks too?

Reply:
Yes, created LOG-8301